Android CI: swift-foundation: error: 'nil' is not compatible with expected element type 'UnsafeMutablePointer<CChar>'

### Description

The Windows toolchain build is [failing](https://ci-external.swift.org/job/swift-main-windows-toolchain/1439/consoleText) while building swift-foundation for Android with

```
swift-foundation/Sources/FoundationEssentials/FileManager/FileOperations+Enumeration.swift:183:60: error: 'nil' is not compatible with expected element type 'UnsafeMutablePointer<CChar>' (aka 'UnsafeMutablePointer<Int8>')
```

### Reproduction

```swift

```


### Expected behavior

No error.

### Environment

Windows CI

### Additional information

_No response_

--- a/Runtimes/Overlay/Android/clang/CMakeLists.txt
+++ b/Runtimes/Overlay/Android/clang/CMakeLists.txt
@@ -30,11 +30,8 @@ target_compile_options(SwiftAndroid INTERFACE
 
 install(FILES
   android.modulemap
+  posix_filesystem.apinotes
+  spawn.apinotes
   SwiftAndroidNDK.h
   SwiftBionic.h
   DESTINATION ${CMAKE_INSTALL_LIBDIR}/swift/${SwiftOverlay_PLATFORM_SUBDIR}/${SwiftOverlay_ARCH_SUBDIR})
-
-install(FILES
-  posix_filesystem.apinotes
-  spawn.apinotes
-  DESTINATION ${CMAKE_INSTALL_LIBDIR}/swift$<$<NOT:$<BOOL:${BUILD_SHARED_LIBS}>>:_static>/apinotes)

